client/plugins/fcc-source-challenges/create-challenge-nodes.js

```diff
@@ -26,13 +28,13 @@ function createChallengeNode(challenge, reporter) {
       Object.assign(
         {},
         {
-          id: challenge.id + ' >>>> ChallengeNode',
           children: [],
           parent: null,
           internal,
           sourceInstanceName: 'challenge'
         },
-        challenge
+        challenge,
+        { id: challenge.id + internal }
```

Comment:
This may be a dumb question, but this appears to create an id like:
```
'bd7158d8c442eddfaeb5bd18[object Object]'
```
because you are concatenating a string with an object.  What is the purpose of this?  Originally it was:
```
id: challenge.id + ' >>>> ChallengeNode'
```

Reply:
No, it's a good question and clearly I need more sleep. It was supposed to be id + type, rather than just id, but it's not.

What I intended to do was fix the order (because previously the assignment order meant the final id was just challenge.id).